### ISSUE TITLE
Yarn ignores misstyped command line arguments

### ISSUE THREAD

**Issue by mprobst** [yarnpkg/yarn#2600]
**Do you want to request a *feature* or report a *bug*?**

A bug.

**What is the current behavior?**

```sh
$ yarn install --parsing-command-line-arguments-is-hard
yarn install v0.19.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[...]
Done in 12.71s.
```

**If the current behavior is a bug, please provide the steps to reproduce.**

`yarn` silently ignores unknown command line arguments. That's a usability problem, i.e. when users mistype an argument, the command silently proceeds but then fails to behave as expected, leaving a puzzled user. Combined with potentially dangerous arguments, this can be a bigger problem. Imagine installing an untrusted package that you'd like to inspect for security reasons, but misstyping `--ignore-scripts`.

**What is the expected behavior?**

```sh
$ yarn install --fried-chicken
yarn install v0.19.1
error Unknown argument "--fried-chicken".
```

**Please mention your node.js, yarn and operating system version.**

Node v5.3.0, Ubuntu.

**Comment by BYK** [comment]
I don't think this is a bug and it is a very good enhancement in terms of safety so marking as high-priority.

**Comment by jotapsa** [comment]
Me and a team of 2 others will attempt to fix this issue

**Comment by jotapsa** [comment]
So basically this issue is due to a issue in commander.js dependency, i created a PR for commander.js https://github.com/tj/commander.js/pull/730 that would fix this, should it get accepted

**Comment by shadowspawn** [comment]
This is likely fixed from Commander v5: https://github.com/tj/commander.js/issues/561
